cache.go

```diff
@@ -0,0 +1,29 @@
+package wazero
+
+import (
+	"context"
+
+	"github.com/tetratelabs/wazero/internal/wasm"
+)
+
+// NewCache returns a new Cache to be passed to RuntimeConfig.
+func NewCache() Cache {
```

Comment:
so the question is if we should add CacheConfig or not
```suggestion
func NewCache(CacheConfig) Cache {
```

Reply:
maybe we can just have WithFileCacheDir() to `type Cache interface` and then we don't need CacheConfig